src/container.jl

```diff
@@ -170,6 +193,22 @@ function effectiveSampleSize(pc::ParticleContainer)
     return inv(sum(abs2, Ws))
 end
 
+"""
+    update_keys!(pc::ParticleContainer)
+
+Create new unique keys for the particles in the ParticleContainer
+"""
+function update_keys!(pc::ParticleContainer, ref::Union{Particle,Nothing}=nothing)
+    # Update keys to new particle ids
+    nparticles = length(pc)
+    n = ref === nothing ? nparticles : nparticles - 1
+    for i in 1:n
+        pi = pc.vals[i]
+        k = split(pi.rng.rng.key)
+        Random.seed!(pi.rng, k[1])
+    end
+end
```

Comment:
I guess we could add an empty `return` or `nothing` at the end to avoid that the last seed or rng is returned.